tests/phpunit/CRM/Contribute/BAO/ContributionTest.php

```diff
@@ -1195,4 +1195,45 @@ public function createContributionWithTax() {
     return array($contribution, $financialAccount);
   }
 
+  /**
+   * function to test card_type and pan truncation.
+   */
+  public function testCardTypeAndPanTruncation() {
+    $contactId = $this->individualCreate();
+    $params = array(
+      'contact_id' => $contactId,
+      'receive_date' => '2016-01-20',
+      'total_amount' => 100,
+      'financial_type_id' => 1,
+      'payment_instrument_id' => 1,
+      'card_type' => 'Visa',
+      'pan_truncation' => 4567,
+    );
+    $contribution = CRM_Contribute_BAO_Contribution::create($params);
+    $lastFinancialTrxnId = CRM_Core_BAO_FinancialTrxn::getFinancialTrxnId($contribution->id, 'DESC');
+    $financialTrxn = $this->callAPISuccessGetSingle(
+      'FinancialTrxn',
+      array(
+        'id' => $lastFinancialTrxnId['financialTrxnId'],
+        'return' => array('card_type', 'pan_truncation'),
+      )
+    );
+    $this->assertEquals(CRM_Utils_Array::value('card_type', $financialTrxn), 1);
+    $this->assertEquals(CRM_Utils_Array::value('pan_truncation', $financialTrxn), 4567);
+    $params = array(
+      'id' => $contribution->id,
+      'pan_truncation' => 2345,
+    );
+    $contribution = CRM_Contribute_BAO_Contribution::create($params);
```

Comment:
@eileenmcnaughton since its a CRM unit test i added BAO method to test if pan truncation is updated correctly. I have also added the same test using api also.